airbyte-workers/src/main/java/io/airbyte/workers/temporal/TemporalClient.java

```diff
@@ -115,10 +146,154 @@ public TemporalClient(final WorkflowClient client, final Path workspaceRoot) {
+    final ConnectionManagerWorkflow connectionManagerWorkflow = getConnectionUpdateWorkflow(connectionId);
+
+    connectionManagerWorkflow.deleteConnection();
+  }
+
+  public void update(final ConnectionUpdate connectionUpdate) throws JsonValidationException, ConfigNotFoundException, IOException {
+    final ConnectionManagerWorkflow connectionManagerWorkflow = getConnectionUpdateWorkflow(connectionUpdate.getConnectionId());
+
+    connectionManagerWorkflow.connectionUpdated();
+  }
+
+  @Value
+  public class ManualSyncSubmissionResult {
+
+    final Optional<String> failingReason;
+    final Optional<Long> jobId;
+
+  }
+
+  public ManualSyncSubmissionResult startNewManualSync(final UUID connectionId) {
+    log.info("Manual sync request");
+    final boolean isWorflowRunning = isWorkflowRunning("connection_updater_" + connectionId);
+
+    if (!isWorflowRunning) {
+      return new ManualSyncSubmissionResult(
+          Optional.of("No scheduler workflow is running for: " + connectionId),
+          Optional.empty());
+    }
+
+    final ConnectionManagerWorkflow connectionManagerWorkflow =
+        getExistingWorkflow(ConnectionManagerWorkflow.class, "connection_updater_" + connectionId);
+    final WorkflowState workflowState = connectionManagerWorkflow.getState();
+
+    if (workflowState.isRunning()) {
+      // TODO Bmoric: Error is running
+      return new ManualSyncSubmissionResult(
+          Optional.of("A sync is already running for: " + connectionId),
+          Optional.empty());
+    }
+
+    connectionManagerWorkflow.submitManualSync();
+
+    do {
+      try {
+        Thread.sleep(10);
+      } catch (final InterruptedException e) {
+        return new ManualSyncSubmissionResult(
+            Optional.of("Didn't managed to start a sync for: " + connectionId),
+            Optional.empty());
+      }
+    } while (!connectionManagerWorkflow.getState().isRunning());
+
+    log.info("end of manual schedule");
+
+    final long jobId = connectionManagerWorkflow.getJobInformation().getJobId();
+
+    return new ManualSyncSubmissionResult(
+        Optional.empty(),
+        Optional.of(jobId));
+  }
+
+  @Value
+  public class NewCancellationSubmissionResult {
+
+    final Optional<String> failingReason;
+    final Optional<Long> jobId;
+
+  }
+
+  public ManualSyncSubmissionResult startNewCancelation(final UUID connectionId) {
+    log.info("Manual sync request");
+
+    final boolean isWorflowRunning = isWorkflowRunning("connection_updater_" + connectionId);
+
+    if (!isWorflowRunning) {
+      return new ManualSyncSubmissionResult(
+          Optional.of("No scheduler workflow is running for: " + connectionId),
+          Optional.empty());
+    }
+
+    final ConnectionManagerWorkflow connectionManagerWorkflow =
+        getExistingWorkflow(ConnectionManagerWorkflow.class, "connection_updater_" + connectionId);
+
+    connectionManagerWorkflow.cancelJob();
+
+    do {
+      try {
+        Thread.sleep(10);
```

Comment:
same here. 

is this meant to be 10 seconds?

Reply:
Done. It is meant to be 10 ms, the reason is that the signal are processed asynchronously by temporal. I wanted the method to return when done so I am waiting that the state query returns the expected value and I add 10 ms as a delay between 2 queries (randomly) in order to have a minimum of a throttle.

Reply:
ok! can we add a comment or make sure the variable name reflects that? 10 ms is a very magic number.

Reply:
Done